videostore/utils.go

```diff
@@ -40,8 +40,32 @@ const (
 	codecH264
 	// CodecH265 represents the H.265 codec type.
 	codecH265
+
+	// maxSegmentLength defines the maximum clip duration in seconds.
+	maxSegmentLength = 30
+
+	// keyFrameIntervalBuffer defines max seconds between keyframes.
+	keyFrameIntervalBuffer = 4
```

Comment:
The viamrtsp videostore path will not always split exactly at 30 seconds. The segmenter needs to wait for keyframe to split so it is possible that we go over configured segment time. This const is used to add an additional buffer length to the initial segment duration estimate during matching.

Reply:
This is good context and worth adding to the above comment

Reply:
Sounds good, added more detailed comment.

Reply:
lets remove this assumption. It can be violated by the camera which we don't control.

Discussed an alternative approach with sean p